tests/src/test/scala/scalaz/FoldableTest.scala

```diff
@@ -281,6 +281,41 @@ object FoldableTest extends SpecLite {
        must_===((l ++ l2).reverse))
   }
 
+  "splitWith on sorted list produces at most 2 elements" ! forAll {
+    (l: List[Int], p: Int => Boolean) =>
+      L.splitWith(l.sortBy(p))(p).size mustBe_< 3
+  }
+
+  "splitWith consistent with partition" ! forAll {
+    (l: List[Int], p: Int => Boolean) =>
+      import scalaz.syntax.std.list._
+      val (trueL, falseL) = l.partition(p)
+      L.splitWith(l.sortBy(p))(p) must_=== List(falseL, trueL).flatMap(_.toNel)
+  }
+
+  "selectSplit: removes all non-satisfying elements" ! forAll {
+    (l: List[Int]) =>
+      L.selectSplit(l)(_ => false).size must_=== 0
+  }
+
+  "selectSplit: keeps all satisfying elements" ! forAll {
+    (l: List[Int]) =>
+      import scalaz.syntax.std.list._
+      L.selectSplit(l)(_ => true) must_=== l.toNel.toList
+  }
+
+  "selectSplit: consistent with partition" ! forAll {
+    (l: List[Int], p: Int => Boolean) =>
+      L.selectSplit(l)(p).flatMap(_.toList) must_=== l.partition(p)._1
+  }
+
+  "selectSplit range consistent with filter" ! forAll {
+    (limit: Byte) =>
+      val l = Math.abs(limit.toInt)
+      val range = List.range(0, l)
```

Comment:
Could you please also take a random lower bound of the range? Otherwise, the predicate below (`_ % 2 != 0`) will never hold for the first element of this range, so it not random enough input to the method.